lib/events/athena/consumer.go

```diff
@@ -398,20 +418,110 @@ func (s *sqsMessagesCollector) downloadEventFromS3(ctx context.Context, payload
+		case eventAndAckID, ok := <-eventsCh:
 			if !ok {
-				return size, nil
+				break eventLoop
+			}
+			pqtEvent, err := auditEventToParquet(eventAndAckID.event)
+			if err != nil {
+				// TODO(tobiaszheller): come back and add some metrics here.
+				c.logger.WithError(err).Error("Could not convert event to parquet format")
+				continue
+			}
+			date := pqtEvent.GetDate()
+			pw := perDateWriter[date]
+			if pw == nil {
+				fw, err := newPerDateFileWriterFn(ctx, date)
+				if err != nil {
+					// While using s3 file writer, error is not used
+					// when creating file writer.
+					return nil, trace.Wrap(err)
+				}
+				pw, err = newParquetWriter(ctx, fw)
+				if err != nil {
+					// Error here means that probably something is wrong with
+					// parquet schema. Returning from fn with error make sense.
+					return nil, trace.Wrap(err)
+				}
+				perDateWriter[date] = pw
 			}
-			size++
-			c.logger.Debugf("Received event: %s %s", eventAndAckID.event.GetID(), eventAndAckID.event.GetType())
+			if err := pw.Write(ctx, *pqtEvent); err != nil {
+				// pw.Write returns error only on flushing operation which
+				// does not happen on every write.
+				// So there is no easy way to say, which even cause trouble and
+				// skip it.
+				// It may happen that one wrong entry will cause whole batch
+				// to write failure. Although it should not happen often because
+				// we are validating message before. If it happen though, whole
+				// batch will go to dead letter.
+				// TODO(tobiaszheller): check how other parquet libs are handling it.
+				// or maybe use Flush explicitly. Need to check performence.
+				return nil, trace.Wrap(err)
+			}
+
+			// Elements are just added to slice here. Acknowledge happens only if whole
+			// writeToS3 method succeed.
+			toDelete = append(toDelete, eventAndAckID.receiptHandle)
 		}
 	}
+
+	for _, pw := range perDateWriter {
+		if err := pw.Close(); err != nil {
+			// Typically there will be data just for one date.
+			// If we are not able to close parquet file, it make sense to retrun
+			// error and retry whole batch again from SQS.
+
+			// TODO(tobiaszheller): verify if broken files are removed from s3.
+			return nil, trace.Wrap(err)
+		}
+	}
+	return toDelete, nil
+}
+
+func newParquetWriter(ctx context.Context, fw source.ParquetFile) (*parquetWriter, error) {
+	pw, err := writer.NewParquetWriter(fw, new(eventParquet), 4)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	pw.CompressionType = parquet.CompressionCodec_SNAPPY
+
+	return &parquetWriter{
+		closer: fw,
+		writer: pw,
+	}, nil
+}
+
+type parquetWriter struct {
+	closer io.Closer
+	writer *writer.ParquetWriter
+}
+
+func (pw *parquetWriter) Write(ctx context.Context, in eventParquet) error {
+	if err := pw.writer.Write(in); err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+func (pw *parquetWriter) Close() error {
+	if err := pw.writer.WriteStop(); err != nil {
+		return trace.Wrap(err)
```

Comment:
it's not needed because [writer on failure will close it,(https://github.com/xitongsys/parquet-go-source/blob/fbbcdea5f512909b970bfc18bb0e14d3ff160cd3/s3v2/s3.go#L258-L269) but we can do it anyway. 
Done in https://github.com/gravitational/teleport/pull/24805/commits/e48a08a859250d62fc54b3ebbf1e28cab76b95c6